src/main/java/com/liberty/basis/PageWrapper.java

```diff
@@ -0,0 +1,157 @@
+package com.liberty.basis;
+
+import org.springframework.data.domain.Page;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Created by user on 16.05.2017.
+ */
+public class PageWrapper<T> {
+    public static final int MAX_PAGE_ITEM_DISPLAY = 5;
+    private Page<T> page;
+    private List<PageItem> items;
+    private int currentNumber;
+    private String url;
+
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(String url) {
+        this.url = url;
+    }
+
+    public PageWrapper(Page<T> page, HttpServletRequest request){
+        this(page, "");
+        LinkedHashMap<String,List<String>> paramsWithoutPagingMap = new LinkedHashMap<>();
+
+        for(Map.Entry<String,String[]> parameter:request.getParameterMap().entrySet()){
+            switch (parameter.getKey()){
+                case "page":
+                case "size":
+                case "_csrf":
+                    break;
+                default:
+                    paramsWithoutPagingMap.put(parameter.getKey(), Arrays.stream(parameter.getValue()).collect(Collectors.toList()));
+            }
+
+        }
+
+
+        MultiValueMap<String,String> paramsWithoutPagingMultiMap = new LinkedMultiValueMap<>(paramsWithoutPagingMap);
+        UriComponents uriComponents = UriComponentsBuilder.fromPath(request.getRequestURI()).queryParams(paramsWithoutPagingMultiMap).build();
+
+        setUrl(uriComponents.toString());
+    }
+
+    public PageWrapper(Page<T> page, String url){
+        this.page = page;
+        this.url = url;
+        items = new ArrayList<PageItem>();
+
+        currentNumber = page.getNumber() + 1; //start from 1 to match page.page
+
+        int start, size;
```

Comment:
should be moved to separate method init(), Do not use login in constructor just call init() from constructor